controller/api/locking.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from models import get_etcd_client
+
+log = logging.getLogger(__name__)
+LOCK_PATH_TEMPLATE = '/deis/controller/locks/{name}'
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+def lock(name, ttl=3600):
+    key = LOCK_PATH_TEMPLATE.format(name=name)
+    value = 'locked'
+    client = get_etcd_client()
```

Comment:
There is no asyncio in python2.
```
 nse deis-controller
bash-4.3# ps
PID   USER     TIME   COMMAND
    1 root       2:34 bash /app/bin/boot
  120 deis       2:17 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
  133 root      13:54 confd -node 10.21.12.124:4001 --confdir /app --log-level error --interval 5
 5050 deis       2:29 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5051 deis       2:39 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5052 deis       2:19 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5053 deis       2:41 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5054 deis       2:33 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5055 deis       2:29 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5056 deis       2:43 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5057 deis       2:30 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
 5058 deis       2:48 {gunicorn} /usr/bin/python /usr/bin/gunicorn -c deis/gconf.py deis.wsgi
24977 root       0:00 sleep 10
24978 root       0:00 bash
24983 root       0:00 ps
bash-4.3# /usr/bin/python --version
Python 2.7.10
```
Im spending some time to find any in memory locking mechanism for python 2.7 and get nothing, if you find any, i will rewrite this to in memory lock.
Also i try to make locking through fcntl fs lock, but there are some problems with this realization due to in one thread sync and multi thread sync.